### PR TITLE
Feature/newtestrunner

### LINKED ISSUE #328: Get duration of non-debug Apex tests
We currently get duration of Apex test execution only when debug is enabled. This should be possible for non-debug tests too, through the tooling API. Query for the field "RunTime" on the Apex test result.

### PR DESCRIPTION
**Pull request by cdcarter**
# Critical Changes
* move RunApexTests from cumulusci.tasks.salesforce, instead offering one RunApexTests in cumulusci.tasks.apex.testrunner
* move AnonymousApexTask to cumulusci.tasks.apex.anon
* move BatchApexWait to cumulusci.tasks.apex.batch
* remove RunApexTestsDebug entirely

# Changes
* RunApexTests not uses the ApexTestResultLimits object to get test limits, as opposed to parsing the debug log.
* Limits usage data now properly only reports limits used between Test.startTest and Test.stopTest, instead of the code used to verify the test.
* saving debug logs directly from a test run is no longer supported.

# Issues Closed
closes #328: get duration of non-debug apex tests

### CODE REVIEW

**Comment by cdcarter** [comment]
Investigating the heroku TFs now, everything passed locally.

@cpolcyn - reviewing 95a24d7 directly (second commit on this branch) should show you the bulk of the actual *changes* instead of just renames/reorg.

File: cumulusci/tasks/apex/testrunner.py [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+
+    # pylint: disable=W0201
+    def _init_class(self):
+        self.classes_by_id = {}
+        self.classes_by_name = {}
+        self.job_id = None
+        self.results_by_class_name = {}
+        self.result = None
+
+    def _decode_to_unicode(self, content):
```
Comment (cpolcyn) [inline cumulusci/tasks/apex/testrunner.py]: Move to cumulusci/utils.py?

File: cumulusci/tasks/apex/testrunner.py [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+            class_name = self.classes_by_id[test_result['ApexClassId']]
+            self.results_by_class_name[class_name][test_result[
+                'MethodName']] = test_result
+            self.counts[test_result['Outcome']] += 1
+        test_results = []
+        class_names = self.results_by_class_name.keys()
+        class_names.sort()
+        for class_name in class_names:
+            message = 'Class: {}'.format(class_name)
+            duration = None  # TODO: Get class total runtime
```
Comment (cpolcyn) [inline cumulusci/tasks/apex/testrunner.py]: Is this value used anywhere else? Wondering if losing this info will have any adverse effect somewhere else.

File: cumulusci/tasks/apex/testrunner.py [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+            self.classes_by_name[test_class['Name']] = test_class['Id']
+            self.results_by_class_name[test_class['Name']] = {}
+        self.logger.info('Queuing tests for execution...')
+        ids = self.classes_by_id.keys()
+        result = self.tooling._call_salesforce(
+            method='POST',
+            url=self.tooling.base_url + 'runTestsAsynchronous',
+            json={'classids': ','.join(str(id) for id in ids)},
+        )
+        if result.status_code != 200:
```
Comment (cpolcyn) [inline cumulusci/tasks/apex/testrunner.py]: Should use httplib.OK for clarity

File: cumulusci/tasks/apex/testrunner.py (hunk 2 above)
Comment (cdcarter) [inline cumulusci/tasks/apex/testrunner.py]: that's what i get for checking in my TODO. it was only logged to the console, so its safe to lose.

**Pull request merged**